rotkehlchen/chain/arbitrum_one/modules/gmx/balances.py

```diff
@@ -0,0 +1,168 @@
+import logging
+from collections import defaultdict
+from itertools import chain
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.ethereum.interfaces.balances import ProtocolWithBalance
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChainID, ChecksumEvmAddress, EvmTokenKind
+
+from .constants import CPT_GMX, GMX_READER, GMX_USD_DECIMALS, GMX_VAULT_ADDRESS
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.arbitrum_one.node_inquirer import ArbitrumOneInquirer
+    from rotkehlchen.chain.ethereum.interfaces.balances import BalancesType
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _chunker(lst: list[Any], chunk_size: int) -> list[list[Any]]:
+    """Used to chunk a list of length (N * chunksize) into a list having N sublists"""
+    return [lst[i:i + chunk_size] for i in range(0, len(lst), chunk_size)]
+
+
+class GmxBalances(ProtocolWithBalance):
+    def __init__(self, database: 'DBHandler', evm_inquirer: 'ArbitrumOneInquirer'):
+        super().__init__(
+            database=database,
+            evm_inquirer=evm_inquirer,
+            counterparty=CPT_GMX,
+            deposit_event_types={(HistoryEventType.DEPOSIT, HistoryEventSubType.DEPOSIT_ASSET)},
+        )
+
+    def _extract_unique_deposits(self) -> dict['ChecksumEvmAddress', set[tuple[str, str, bool]]]:
+        """
+        fetch deposit events and remove duplicate positions. Since a user can modify the same
+        position increasing or decreasing the collateral we want to remove duplicates to make
+        the less queries possible.
+        """
+        addresses_events = self.addresses_with_activity(
+            products=None,
+            event_types=self.deposit_event_types,
+        )
+        unique_searches = {}
```

Comment:
I moved this part of the logic to a function and then renamed to deposits, didn't change this variable. The reason is because I create here the tuples that will be used to search the user's positions later but using deposits is clearer here